com.unity.render-pipelines.high-definition/Runtime/Lighting/LightLoop/scrbound.compute

```diff
@@ -1,538 +1,740 @@
-                                // However, worst case the plane that would be skipped if such an extreme case ever happened would be backplane
-                                // clipping gets skipped which doesn't cause any errors.
+/* ------------------------------ Utilities --------------------------------- */
 
+// Returns the location of the N-th set bit starting from the lowest order bit and working upward.
+// Slow implementation - do not use for large bit sets.
+// Could be optimized - see https://graphics.stanford.edu/~seander/bithacks.html
+uint NthBitLow(uint value, uint n)
+{
+    uint b = -1;                                    // Consistent with the behavior of firstbitlow()
+    uint c = countbits(value);
+
+    if (n < c)                                      // Validate inputs
+    {
+        uint r = n + 1;                             // Compute the number of remaining bits
 
-// LDS (2496 bytes)
-groupshared float posX[MAX_PNTS*8*2];
-groupshared float posY[MAX_PNTS*8*2];
-groupshared float posZ[MAX_PNTS*8*2];
-groupshared float posW[MAX_PNTS*8*2];
-groupshared unsigned int clipFlags[48];
+        do
+        {
+            uint f = firstbitlow(value >> (b + 1)); // Find the next set bit
+            b += f + r;                             // Make a guess (assume all [b+f+1,b+f+r] bits are set)
+            c = countbits(value << (32 - (b + 1))); // Count the number of bits actually set
+            r = (n + 1) - c;                        // Compute the number of remaining bits
+        } while (r > 0);
+    }
 
+    return b;
+}
 
-unsigned int GetClip(const float4 P);
-int ClipAgainstPlane(const int iSrcIndex, const int iNrSrcVerts, const int subLigt, const int p);
-void CalcBound(out bool2 bIsMinValid, out bool2 bIsMaxValid, out float2 vMin, out float2 vMax, float4x4 InvProjection, float3 pos_view_space, float r);
+float4x4 Translation4x4(float3 d)
+{
+    float4x4 M = k_Identity4x4;
 
-#include "Packages/com.unity.render-pipelines.high-definition/Runtime/Lighting/LightLoop/LightingConvexHullUtils.hlsl"
+    M._14_24_34 = d; // Last column
 
+    return M;
+}
 
-[numthreads(NR_THREADS, 1, 1)]
-void SCRAABBGEN(uint threadID : SV_GroupIndex, uint3 u3GroupID : SV_GroupID)
+// Scale followed by rotation (scaled axes).
+float3x3 ScaledRotation3x3(float3 xAxis, float3 yAxis, float3 zAxis)
 {
-    uint groupID = u3GroupID.x;
-    uint eyeIndex = u3GroupID.y; // currently, can only be 0 or 1
+    float3x3 R = float3x3(xAxis, yAxis, zAxis);
+    float3x3 C = transpose(R); // Row to column
 
-    // The g_ is preserved in order to make cross-pipeline (FPTL) updates easier
-    float4x4 g_mInvProjection = g_mInvProjectionArr[eyeIndex];
-    float4x4 g_mProjection = g_mProjectionArr[eyeIndex];
+    return C;
+}
 
-    //uint vindex = groupID * NR_THREADS + threadID;
-    unsigned int g = groupID;
-    unsigned int t = threadID;
+float3x3 Invert3x3(float3x3 R)
```

Comment:
Feels like this would be useful even outside of this context on its own hlsl file (or already existing common file?) along few of the other generic matrix ops in this file.

Reply:
Sure. We just need to decide what to move to the core.

Reply:
We could add those in a "math section" in common.hlsl. Or alternatively create a matrix.hlsl file?

Reply:
We should avoid creating more files. I propose we move this to Common once there is more than 1 use case.

Reply:
same answer than before, nobody will ever know that such a function exist as it is "hidden" in particular file.
Agree to not add more files, so we could move it to common.hlsl, it is a frequent enough function to be there.

Reply:
I will see whether it makes sense to share some code after I am done with Z-binning.